discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -234,48 +236,53 @@ def process_create_userbank_instruction(
     for memo in memos:
         logger.debug(f"index_user_bank.py | MEMO {memo}")
         try:
             content_metadata = json.loads(memo)
-            if "type" in content_metadata and "id" in content_metadata:
-                premium_conditions = None
-                if PurchaseType[content_metadata["type"].lower()] == PurchaseType.track:
-                    results = (
-                        session.query(Track.premium_conditions, Track.owner_id)
+            logger.debug(f"index_user_bank.py | Found JSON in memo: {content_metadata}")
+            if (
+                "type" in content_metadata
+                and "id" in content_metadata
+                and "blocknumber" in content_metadata
+            ):
+                # TODO: Wait for blocknumber to be indexed by ACDC
+                (id, type, _) = content_metadata["id"], content_metadata["type"], content_metadata["blocknumber"]
+                price = None
+                splits = None
+                if PurchaseType[type.lower()] == PurchaseType.track:
+                    result: TrackPriceHistory = (
+                        session.query(TrackPriceHistory)
                         .filter(
-                            Track.track_id == content_metadata["id"],
-                            Track.is_current.in_(True, False),
-                            Track.premium_conditions["usdc_purchase"] != None,
-                            Track.premium_conditions["usdc_purchase"][
-                                "slot"
-                            ].astext.cast(BigInteger)
-                            <= slot,
+                            TrackPriceHistory.track_id == id,
+                            TrackPriceHistory.block_timestamp < timestamp
                         )
+                        .order_by(desc(TrackPriceHistory.block_timestamp))
                         .first()
                     )
+                    if result is not None:
+                        price = result.total_price_cents
+                        splits = result.splits
                 else:
                     logger.error(
                         f"index_user_bank.py | Unknown content type {content_metadata['type']}"
                     )
-                if results:
-                    premium_conditions, owner_id = results
+                if price is not None and splits is not None:
                     return {
                         "type": PurchaseType[content_metadata["type"].lower()],
                         "id": content_metadata["id"],
-                        "owner_id": owner_id,
-                        "price": int(premium_conditions["usdc_purchase"]["price"])
-                        * USDC_PER_USD_CENT,
+                        "price": price * USDC_PER_USD_CENT,
+                        "splits": splits
                     }
                 else:
                     logger.error(
```

Comment:
Nit: log lies now, this is if we can't find TrackPriceHistory for the track

Reply:
Good catch!